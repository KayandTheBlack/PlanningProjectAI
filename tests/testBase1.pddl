(define (problem autogen)
  (:domain worldsEndAndBeyond)

  (:objects
    Origin cityA cityB cityC cityD cityE cityF cityG cityH cityI cityJ cityK cityL cityM cityN cityO - city
    hotelA_A hotelA_B hotelA_C hotelB_A hotelB_B hotelB_C hotelC_A hotelC_B hotelC_C hotelD_A hotelD_B hotelD_C hotelE_A hotelE_B hotelE_C hotelF_A hotelF_B hotelF_C hotelG_A hotelG_B hotelG_C hotelH_A hotelH_B hotelH_C hotelI_A hotelI_B hotelI_C hotelJ_A hotelJ_B hotelJ_C hotelK_A hotelK_B hotelK_C hotelL_A hotelL_B hotelL_C hotelM_A hotelM_B hotelM_C hotelN_A hotelN_B hotelN_C hotelO_A hotelO_B hotelO_C - hotel
  )

  (:init
    (currentLocation Origin)

    (hotelAt hotelA_A cityA)
    (hotelAt hotelA_B cityA)
    (hotelAt hotelA_C cityA)
    (hotelAt hotelB_A cityB)
    (hotelAt hotelB_B cityB)
    (hotelAt hotelB_C cityB)
    (hotelAt hotelC_A cityC)
    (hotelAt hotelC_B cityC)
    (hotelAt hotelC_C cityC)
    (hotelAt hotelD_A cityD)
    (hotelAt hotelD_B cityD)
    (hotelAt hotelD_C cityD)
    (hotelAt hotelE_A cityE)
    (hotelAt hotelE_B cityE)
    (hotelAt hotelE_C cityE)
    (hotelAt hotelF_A cityF)
    (hotelAt hotelF_B cityF)
    (hotelAt hotelF_C cityF)
    (hotelAt hotelG_A cityG)
    (hotelAt hotelG_B cityG)
    (hotelAt hotelG_C cityG)
    (hotelAt hotelH_A cityH)
    (hotelAt hotelH_B cityH)
    (hotelAt hotelH_C cityH)
    (hotelAt hotelI_A cityI)
    (hotelAt hotelI_B cityI)
    (hotelAt hotelI_C cityI)
    (hotelAt hotelJ_A cityJ)
    (hotelAt hotelJ_B cityJ)
    (hotelAt hotelJ_C cityJ)
    (hotelAt hotelK_A cityK)
    (hotelAt hotelK_B cityK)
    (hotelAt hotelK_C cityK)
    (hotelAt hotelL_A cityL)
    (hotelAt hotelL_B cityL)
    (hotelAt hotelL_C cityL)
    (hotelAt hotelM_A cityM)
    (hotelAt hotelM_B cityM)
    (hotelAt hotelM_C cityM)
    (hotelAt hotelN_A cityN)
    (hotelAt hotelN_B cityN)
    (hotelAt hotelN_C cityN)
    (hotelAt hotelO_A cityO)
    (hotelAt hotelO_B cityO)
    (hotelAt hotelO_C cityO)

    (flight Origin cityA)
    (flight Origin cityB)
    (flight Origin cityC)
    (flight Origin cityD)
    (flight Origin cityE)
    (flight Origin cityF)
    (flight Origin cityG)
    (flight Origin cityH)
    (flight Origin cityI)
    (flight Origin cityJ)
    (flight Origin cityK)
    (flight Origin cityL)
    (flight Origin cityM)
    (flight Origin cityN)
    (flight Origin cityO)
    (flight cityA cityC)
    (flight cityA cityF)
    (flight cityA cityN)
    (flight cityB cityC)
    (flight cityB cityF)
    (flight cityB cityL)
    (flight cityC cityA)
    (flight cityC cityJ)
    (flight cityC cityM)
    (flight cityD cityC)
    (flight cityD cityE)
    (flight cityD cityJ)
    (flight cityE cityD)
    (flight cityE cityH)
    (flight cityE cityI)
    (flight cityF cityE)
    (flight cityF cityI)
    (flight cityF cityJ)
    (flight cityG cityB)
    (flight cityG cityK)
    (flight cityG cityO)
    (flight cityH cityD)
    (flight cityH cityL)
    (flight cityH cityM)
    (flight cityI cityA)
    (flight cityI cityC)
    (flight cityI cityM)
    (flight cityJ cityI)
    (flight cityJ cityL)
    (flight cityJ cityO)
    (flight cityK cityA)
    (flight cityK cityD)
    (flight cityK cityO)
    (flight cityL cityF)
    (flight cityL cityJ)
    (flight cityL cityK)
    (flight cityM cityA)
    (flight cityM cityC)
    (flight cityM cityL)
    (flight cityN cityB)
    (flight cityN cityD)
    (flight cityN cityL)
    (flight cityO cityD)
    (flight cityO cityJ)
    (flight cityO cityM)

    (= (citiesVisited) 0)

  )

  (:goal
    (>= (citiesVisited) 12)
  )


)
