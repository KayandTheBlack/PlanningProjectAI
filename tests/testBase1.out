
ff: parsing domain file
domain 'WORLDSENDANDBEYOND' defined
 ... done.
ff: parsing problem file
problem 'AUTOGEN' defined
 ... done.


no metric specified. plan length assumed.

checking for cyclic := effects --- OK.

ff: search configuration is  best-first on 1*g(s) + 5*h(s) where
    metric is  plan length

advancing to distance:   12
                         11
                         10
                          9
                          8
                          7
                          6
                          5
                          4
                          3
                          2
                          1
                          0

ff: found legal plan as follows

step    0: FLYANDSTAY ORIGIN CITYO HOTELO_A
        1: FLYANDSTAY CITYO CITYM HOTELM_A
        2: FLYANDSTAY CITYM CITYL HOTELL_A
        3: FLYANDSTAY CITYL CITYK HOTELK_A
        4: FLYANDSTAY CITYK CITYD HOTELD_A
        5: FLYANDSTAY CITYD CITYJ HOTELJ_A
        6: FLYANDSTAY CITYJ CITYI HOTELI_A
        7: FLYANDSTAY CITYI CITYC HOTELC_A
        8: FLYANDSTAY CITYC CITYA HOTELA_A
        9: FLYANDSTAY CITYA CITYN HOTELN_A
       10: FLYANDSTAY CITYN CITYB HOTELB_A
       11: FLYANDSTAY CITYB CITYF HOTELF_A
     

time spent:    0.00 seconds instantiating 180 easy, 0 hard action templates
               0.00 seconds reachability analysis, yielding 47 facts and 180 actions
               0.00 seconds creating final representation with 46 relevant facts, 1 relevant fluents
               0.00 seconds computing LNF
               0.00 seconds building connectivity graph
               0.00 seconds searching, evaluating 38 states, to a max depth of 0
               0.00 seconds total time

